### ISSUE TITLE
Multiple recipes in the same PR

### ISSUE THREAD

**Issue by sebastian-luna-valero** [conda-forge/staged-recipes#3424]
Hello,

I would like to add multiple R packages, can I do that in the same pull request?

Best regards,
Sebastian

**Comment by brainstorm** [comment]
I'm new here, but I'm afraid you'll have to break them out just in case, see what happened on my own recent PR and CI logsize limits:

https://github.com/conda-forge/staged-recipes/pull/3421

**Comment by jakirkham** [comment]
TL;DR Can you? Yes. Should you? Probably not.

Within `staged-recipes` we use a tool called [`conda-build-all`]( https://github.com/conda-tools/conda-build-all ). This tool scans through the recipes, figures out what needs to be built, figures out what combinations needed to build, and figures out what order to build everything in. So this tool can handle building multiple recipes correctly.

That said, as noted by @brainstorm, there are CI resource limitations. Though, and this is arguably more important, there are practical limits on what reviewers are able/willing to do. A large PR with many recipes makes it more difficult to review. If the recipes make it through these two constraints and get merged, race conditions amongst the different feedstocks CIs kick in. Meaning work by you and/or core to restart them in such a way to build everything.

None of this is to say that one can't add multiple recipes in a single PR. One certainly can do this and it can work. However there are trade-offs that may make the less desirable. In practice, having a couple of recipes (e.g. 2 or 3) in a PR normally works fine.

**Comment by sebastian-luna-valero** [comment]
Make sense, many thanks!

Would it be a good idea to copy and paste @jakirkham's comments somewhere visible to contributors of the `staged-recipes` repository?

**Comment by jakirkham** [comment]
Sure could go in the docs under guidelines. Would you like to do a PR?

**Comment by sebastian-luna-valero** [comment]
Sorry, I can't see a `guideline` section in the docs for `staged-recipes`. Did you mean `FAQ`?

**Comment by jakirkham** [comment]
Please see this [file]( https://github.com/conda-forge/conda-forge.github.io/blob/77360dd9063c29458fdd21daea5dc8a2be08337c/src/guidelines.rst ).

**Comment by sebastian-luna-valero** [comment]
Thanks! Submitted https://github.com/conda-forge/conda-forge.github.io/pull/424
